### ISSUE TITLE
Add repl command with test artifacts on path, corrensponding sbt command Test/console

### ISSUE THREAD

**Issue by bjornregnell** [VirtusLab/scala-cli#2953]
**Is your feature request related to a problem? Please describe.**
My munit test has a bug.

**Describe the solution you'd like**
I want to debug my munit test in repl using scala-cli.

**Describe alternatives you've considered**
Use the sbt command `Test/console`

**Additional context**
Found nothing about this when reading `scala-cli repl --help-full`

**Comment by SethTisue** [comment]
if this is actually supported somehow, I can't find it either in the doc or via experiment (e.g. `scala-cli repl --test .` is the syntax I would expect and doesn't work)

(and if it _isn't_ supported, it seems like a pretty big hole to me)

**Comment by bjornregnell** [comment]
> seems like a pretty big hole to me

I agree. This seems like a feature that should be assigned the "missing feature" label.

**Comment by Gedochao** [comment]
> (and if it _isn't_ supported, it seems like a pretty big hole to me)

It seems it isn't supported at the moment, so the feature request is valid. 
Shouldn't be too hard to deliver (hopefully).

**Comment by bjornregnell** [comment]
I guess users may try this from two angles: 1) they use the test command but want repl and maybe they try `scala-cli test . --repl`  or 2) they use the repl command but want to include test artifacts and maybe they try `scala-cli repl . --test` or similar.

Perhaps both ways should be supported? Or of only one, I guess 2) is the most natural as it is the repl that is started...

**Comment by SethTisue** [comment]
I agree 2 is more natural (but `repl --test .`, not `repl . --test`). I don't see a need to support 1 (and having multiple ways to do things is a cost).

**Comment by bjornregnell** [comment]
Hmm. You mean that the "test" string is passed as argument to the repl "app"?

I'd rather expect that there is an argument to scala-cli itself that adds extra stuff to the path, similar to `--workspace` or `--js-linker-path` that are args to the `compile` command. So if no path is given to `--test` then it takes `src/test/scala` as default.

**Comment by bjornregnell** [comment]
Aha. I thought you meant `repl -- test` (with a space) but I see now that its just `--test`.

But we need the `.` or `dir1/file.scala` as arg to put the ordinary code on path so I think it should be `scala-cli repl . --test [optional alt test path]` as `scala-cli repl` without any path is not adding anything to the class path. Or maybe I'm missing something?

**Comment by bjornregnell** [comment]
Aha I now see the actual dot at the end. (Sorry for my confusion.) That makes sense to me as well!

**Comment by bjornregnell** [comment]
Anyway I guess it is not all that obvious in which order and how to add test artifacts in non-default dirs.

**Comment by Gedochao** [comment]
We already have `compile --test`, while we don't have `test --compile` 😅 
`repl --test` is what I plan to go with, I just need to find time to implement it.

**Comment by bjornregnell** [comment]
Great. But with the dot or not for "current dir"? And where to put the dot?

It seems like `scala-cli compile . --test` accepts the dot both before and after the `--test` option so `scala-cli compile --test .` also seems to work.

**Comment by Gedochao** [comment]
> But with the dot or not for "current dir"?

`.` (the dot) is an input, which stands for the current working directory.
Launching `repl --test` without any inputs would mean you're launching the repl while including the test scope, but the test scope will be empty, since there are no inputs.

TL;DR yeah, the dot is necessary.

> And where to put the dot?

As with any other case, `.` is treated as regular input, which can be passed anywhere after the subcommand name.
You can even pass inputs interchangeably with options, just take care not to pass it as an option argument by accident.
For example, the following is a valid command:
```bash
scala-cli repl -S 3.4.2 . --jvm 22
```

**Comment by bjornregnell** [comment]
Makes sense! Thanks for clarification. Liberal rules, with some gotchas...

**Comment by bjornregnell** [comment]
So if I write `scala-cli repl someDir --test` does that mean that tests have to be in `someDir/src/test/scala` or is scala-cli looking at any sub-dir called `test` for test code? Can I have my test code in a custom location?

**Comment by Gedochao** [comment]
> So if I write `scala-cli repl someDir --test` does that mean that tests have to be in `someDir/src/test/scala` or is scala-cli looking at any sub-dir called `test` for test code? Can I have my test code in a custom location?

It needs to be identified as a test source, as per this doc:
https://scala-cli.virtuslab.org/docs/commands/test#test-sources
